data/projects/implement_aspects.md

```diff
@@ -1,29 +1,63 @@
+all the functionality provided by various tools inside coala. Beside that, coala
+tries to provide a full set of option for those tools. And also there is a
+possibility that the same type of checking is executed by different tools to
+the same line of code, resulting in duplicated report.
+
 In the future, coala will choose the bears to run for the user based on what
 aspects of a programming language he wants to analyze. This project should
 realize this.
 
-The aspects concept exists as
-[cEP-0005](https://github.com/coala/cEPs/blob/master/cEP-0005.md) and it is
+We could say "aspects" is a what kind of analysis we want to run over the code.
+For example, "CodeCloneDetection" is a kind of aspects that will detect
+portion of code that have exact duplicates somewhere in the program and warn the
+user about it. Now if user want to detect duplicated codes, instead of stating
+what type of bear coala must use (which require user to know list of bears/tools
+and configure their option), they can simply add the CodeCloneDetection aspect
+and coala will figures out what bears must be used and how to use them.
+
+Further technical explanation about aspects is explained in
+[cEP-0005](https://github.com/coala/cEPs/blob/master/cEP-0005.md). It is
 already possible to define aspects and use them in bears - however this has no
 effect because coala doesn't understand what to do with them yet.
 
-This project is about implementing aspects support in coala. coala will have to
-choose the right bears on the basis of aspects and show the right results.
-It should be able to ignore aspects just like bear ignoring works right now.
+This project goal is to implement aspects support in coala.
+
+coala will have to choose the right bears on the basis of aspects and show the
+right results. It should be able to ignore aspects just like bear ignoring
+works right now.
+
+The difference between this project and "Convert Bears to Aspects" is that the
+other project focus on defining aspects on *each bears* while this project
+implement how coala core is handling bears that use aspects.
+
+References
+
+* Aspects concept based on
+  [cEP-0005](https://github.com/coala/cEPs/blob/master/cEP-0005.md)
+* Proposal on how new bears could be designed
+  [cEP-0002](https://github.com/coala/cEPs/blob/master/cEP-0002.md)
+* Current aspects prototype
+  [coala/aspect-docs](https://github.com/coala/aspect-docs)
 
 #### Milestones
 
-##### GSOC 2017 COMMUNITY BONDING
+##### PREPARATION/BONDING
 
 * A solid plan for the implementation and the exact features exist.
 * Any needed changes to [cEP-0005](https://coala.io/cep5) have been merged.
 
-##### GSOC 2017 MIDTERM
+##### CODING PHASE 1
 
 * The new aspect based configuration file works.
 * The new aspect configuration file is documented properly.
 
-##### GSOC 2017 FINAL
+##### CODING PHASE 2
+
+* Ensure backward compatibility with old bears and configuration file.
+
+##### CODING PHASE 3
 
```

Comment:
Aspects should work here. So coala understands aspects config files, chooses the right bears accordingly and returns only the relevant results in case tools do more than needed.